### PR TITLE
Watch permissions

### PR DESCRIPTION
**Pull request by zpytela**
Introduce new permissions for the classes file and filesystem: watch watch_mount watch_reads watch_sb watch_with_perm
```
0c97d3925 Allow policykit various watch permissions
d620b3848 Allow dnsmasq watch systemd-resolved pid directories
49d0a1a3b Allow session_bus_type watch directories in /etc/dbus-1 and /usr
3d73abf0c Add interfaces to watch various files and directories
cd255e10f Allow nscd watch files and directories in /etc
0c384b2d9 Allow init_t watch /var/run/systemd/ask-password
0ce9ba052 Allow firewalld watch directories in /etc/firewalld
75080145b Allow login_userdomain watch various files/symlinks/dirs
c5db9d6e1 Add various systemd-related watches
beba762f1 Allow systemd-dbusd watch directories in /etc/dbus-1 and /usr
c9f17432e Allow sssd watch various files/symlinks/dirs
84d08c999 Allow syslogd watch its private runtime dirs and generic log dirs
35eb390f0 Allow init_t watch init_var_run_t directories
59182f053 Allow getty_t watch getty_var_run_t files
40bdc69de Disable automatic watch permissions for init_t
709de9cb9 Allow init_t to watch mount directories
7dc5adde6 Allow init_t to watch localization symlinks
843367009 Allow init_t to watch cgroup files
4a4761a66 Add basic watch patterns
c0135bbb2 Disable automatic watch permissions for unconfined domains
72ff7700a Set up scaffolding for the new watch permissions
32d6ecda5 Add new watch* permissions
```

### CODE REVIEW

**Comment by zpytela** [comment]
This is a draft PR. It is not expected to be used for other then testing purpose to gather additional denials.
Further notes:
- temporarily, there are generic auditallow rules for the watch permissions to keep the system working
- additional permissions are known missing, they will be added later to make this PR more comprehensible (abrt*_t colord_t blkmapd_t rpcd_t gssd_t avahi_t vdagent_t)
There are copr-builds available on regular basis, the latest one confirmed working is selinux-policy-3.14.7-14.205.watch.fc34, with auditallow rules. Use the following commands:
```
dnf copr enable zpytela/selinux-policy
dnf update selinux policy
```

**Comment by zpytela** [comment]
Just added
```
739e0a897 Allow init_t watch_reads init_var_run_t directories
```
I am aware the watch_reads perm may be later added as an additional one to an existing interface. This can happen to other commits as well, and anyway the complete patchset will be a subject of reviewing and rearranging once majority of permissions is gathered

There is now `selinux-policy-3.14.7-14.206.watch.fc34` copr-build available.

**Comment by WOnder93** [comment]
How about we create a `watch-permissions` branch directly in this repo so that anyone can send PRs against that branch? Then we can also set up a COPR repo to rebuild the latest version automatically.

(Any in-progress PRs will need careful rebasing whenever the branch would be rebased or some commits squashed, but that's not a big deal IMHO.)

**Comment by zpytela** [comment]
The latest copr-build is selinux-policy-3.14.7-16.301.watch.fc34. System boots and runs without a denial, but not all services were examined.
Permissions for systemd are enumeated although allowing watch for file_type (or e. g. non_security_file_type) is still under consideration.

**Comment by zpytela** [comment]
I have just grouped and reordered existing commits for better clarity. I'd appreciate a review now.

**Comment by zpytela** [comment]
The latest copr-build is selinux-policy-3.14.7-16.305.watch.fc34. System boots without an AVC denial.

System also boots with the mls policy without watch-related AVC denial. Some additional denials will likely pop up for domains like sysadm_dbusd_t.

**Review by Koncpa — APPROVED** [review]

File: policy/modules/system/systemd.if [hunk 1]
```diff
@@ -1047,6 +1104,24 @@ interface(`systemd_relabelto_fifo_file_passwd_run',`
 	allow $1 systemd_passwd_var_run_t:fifo_file relabelto;
 ')
 
+########################################
+## <summary>
+##	Watch
```
Comment (Koncpa) [inline policy/modules/system/systemd.if]: Maybe write more about this interface in summary

File: policy/modules/system/miscfiles.if [hunk 2]
```diff
@@ -536,6 +536,31 @@ interface(`miscfiles_read_localization',`
 	allow $1 locale_t:file map;
 ')
 
+########################################
+## <summary>
+##	Allow process to watch localization symlnks.
```
Comment (Koncpa) [inline policy/modules/system/miscfiles.if]: *symlinks

**Comment by zpytela** [comment]
Fixed, thank you.

**Comment by zpytela** [comment]
I am going to merge this PR, all tests results are promising.

**Pull request merged**